tasks/tasks/causemos_processors_test.py

```diff
@@ -0,0 +1,415 @@
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v2_control_data():
+
+    pca_details = (
+        numpy.array(
+            [[1.13900827e-03, 4.01488796e-02, 7.06536188e-01,
+              7.06536188e-01, 0.00000000e+00],
+             [7.13923827e-01, 6.99690962e-01, 1.93044911e-02,
+              1.93044911e-02, 0.00000000e+00],
+             [7.00222444e-01, 7.13316638e-01, 2.08315041e-02,
+              2.08315041e-02, 0.00000000e+00],
+             [2.11047685e-17, 2.19081849e-17, 7.07106781e-01,
+              7.07106781e-01, 0.00000000e+00],
+             [0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+              0.00000000e+00, 1.00000000e+00]]
+        ),
+
+        5,
+
+        numpy.array(
+            [5.00405570e-01, 2.60246340e-01, 2.39348091e-01,
+             2.51310581e-32, 0.00000000e+00]
+        )
+    )
+
+    weights = pca_to_weights_v2(pca_details)
+
+    rounded_as_percent = list(map(lambda i: f"{roundToFour(i) * 100}%", weights))
+
+    assert rounded_as_percent == ['32.45%', '33.32%', '17.11%', '17.11%', '0.0%']
+
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v2_control_data():
+
+    pca_details = (
+        numpy.array(
+            [[1.13900827e-03, 4.01488796e-02, 7.06536188e-01,
+              7.06536188e-01, 0.00000000e+00],
+             [7.13923827e-01, 6.99690962e-01, 1.93044911e-02,
+              1.93044911e-02, 0.00000000e+00],
+             [7.00222444e-01, 7.13316638e-01, 2.08315041e-02,
+              2.08315041e-02, 0.00000000e+00],
+             [2.11047685e-17, 2.19081849e-17, 7.07106781e-01,
+              7.07106781e-01, 0.00000000e+00],
+             [0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+              0.00000000e+00, 1.00000000e+00]]
+        ),
+
+        5,
+
+        numpy.array(
+            [5.00405570e-01, 2.60246340e-01, 2.39348091e-01,
+             2.51310581e-32, 0.00000000e+00]
+        )
+    )
+
+    weights = pca_to_weights_v2(pca_details)
+
+    rounded_as_percent = list(map(lambda i: f"{roundToFour(i) * 100}%", weights))
+
+    assert rounded_as_percent == ['32.45%', '33.32%', '17.11%', '17.11%', '0.0%']
+
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v2_real_index_data():
+
+    rd = iteration_func(index_model_object)
+    datasets_ids = download_datasets(rd)
+
+    out = build_synthetic_dataset(datasets_ids, rd)
+
+    normalized = normalize_synthetic_dataset(out)
+
+    pca_details = perform_pca(normalized)
+
+    weights = pca_to_weights_v2(pca_details)
+
+    rounded_as_percent = list(map(lambda i: f"{roundToFour(i) * 100}%", weights))
+
+    assert rounded_as_percent == ['2.76%', '3.88%', '3.65%', '5.82%', '5.53%', '5.8500000000000005%', '0.0%', '3.6799999999999997%', '6.7299999999999995%', '5.35%', '6.63%', '5.33%', '3.44%', '3.4000000000000004%', '6.18%', '6.09%', '6.99%', '6.13%', '5.55%', '3.2800000000000002%', '3.71%']
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v1_control_data():
```

Comment:
@brandomr: This is the original pca-to-weight v1 algorithm which is currently deployed (not updated matrix one nor the one you suggested in slack). It's running aginst pca output for the control dataset.

The other following test is the same v1 nbut running against the real nd gain.